src/test/rgw_directory_test.cc

```diff
@@ -0,0 +1,108 @@
+#include "rgw_directory.h"
+#include <cpp_redis/cpp_redis>
+#include <iostream>
+#include <string>
+#include "gtest/gtest.h"
+
+using namespace std;
+
+string portStr;
+string redisHost = "";
+string host = "127.0.0.1";
+
+class DirectoryFixture: public ::testing::Test {
+  protected:
+    virtual void SetUp() {
+      blk_dir = new RGWBlockDirectory(host, stoi(portStr));
+      c_blk = new cache_block();
+
+      c_blk->hosts_list.push_back(redisHost);
+      c_blk->size_in_bytes = 123; 
+      c_blk->c_obj.bucket_name = "testBucket";
+      c_blk->c_obj.obj_name = "samoid";
```

Comment:
you might want to add a `string object_name = testoid` as a global variable since you're using the value of `"samoid"` many times in this file.

Reply:
The same goes for the `bucket_name` and `size_in_bytes`